chia/rpc/farmer_rpc_api.py

```diff
@@ -1,11 +1,66 @@
-from typing import Any, Callable, Dict, List, Optional
+import dataclasses
+import operator
+from typing import Any, Callable, Dict, List, Optional, Union
+
+from typing_extensions import Protocol
 
 from chia.farmer.farmer import Farmer
+from chia.plot_sync.receiver import Receiver
+from chia.protocols.harvester_protocol import Plot
 from chia.types.blockchain_format.sized_bytes import bytes32
 from chia.util.byte_types import hexstr_to_bytes
+from chia.util.misc import KeyValue
+from chia.util.paginator import Paginator
+from chia.util.streamable import dataclass_from_dict
 from chia.util.ws_message import WsRpcMessage, create_payload_dict
 
 
+class PaginatedRequestData(Protocol):
+    peer_id: bytes32
+    page: int
+    page_size: int
+
+
+@dataclasses.dataclass
+class PlotInfoRequestData:
+    peer_id: bytes32
+    page: int
+    page_size: int
+    filter: List[KeyValue] = dataclasses.field(default_factory=list)
+    sort_key: str = "filename"
+    reverse: bool = False
+
+
+@dataclasses.dataclass
+class PlotPathRequestData:
+    peer_id: bytes32
+    page: int
+    page_size: int
+    filter: List[str] = dataclasses.field(default_factory=list)
+    reverse: bool = False
+
+
+def paginated_plot_request(source: List[Any], request: PaginatedRequestData) -> Dict[str, object]:
+    paginator: Paginator = Paginator(source, request.page_size)
+    return {
+        "peer_id": request.peer_id.hex(),
+        "page": request.page,
+        "page_count": paginator.page_count(),
+        "total_count": len(source),
+        "plots": paginator.get_page(request.page),
+    }
+
+
+def is_filter_match(plot: Union[Plot, Dict[str, Any]], filter_item: KeyValue) -> bool:
+    if isinstance(plot, Plot):
+        plot_attribute = getattr(plot, filter_item.key)
+    else:
+        plot_attribute = plot[filter_item.key]
+    none_match = filter_item.value is None and plot_attribute is None
+    attribute_match = filter_item.value is not None and filter_item.value in str(plot_attribute)
+    return none_match or attribute_match
```

Comment:
```suggestion
    if filter_item.value is None:
        return plot_attribute is None
    else:
        return filter_item.value in str(plot_attribute)
```

Reply:
Also handled in da6f90b9a08d6fb8570acad95caed03ba9bf8bd9